src/ui/src/core/navigation.ts

```diff
@@ -0,0 +1,68 @@
+type ParsedHash = {
+	pageKey?: string;
+	routeVars: Record<string, string>;
+};
+
+const hashRegex = /^((?<pageKey>[^/]*))?(\/(?<routeVars>.*))?$/;
+const routeVarRegex = /^(?<key>[^=]+)=(?<value>.*)$/;
+
+export function getParsedHash(): ParsedHash {
+	const docHash = document.location.hash.substring(1);
+	const hashMatchGroups = docHash.match(hashRegex)?.groups;
+
+	if (!hashMatchGroups) return { pageKey: undefined, routeVars: {} };
+
+	const routeVars: Record<string, string> = {};
+	const pageKey = hashMatchGroups?.pageKey
+		? decodeURIComponent(hashMatchGroups.pageKey)
+		: undefined;
+
+	const routeVarsSegments = hashMatchGroups.routeVars?.split("&") ?? [];
+	routeVarsSegments.forEach((routeVarSegment) => {
+		const matchGroups = routeVarSegment.match(routeVarRegex)?.groups;
+		if (!matchGroups) return;
+		const { key, value } = matchGroups;
+		const decodedKey = decodeURIComponent(key);
+		const decodedValue = decodeURIComponent(value);
+		routeVars[decodedKey] = decodedValue;
```

Comment:
This didn't pose an immediate risk as it'd require someone to alter the URL to get a denial of service (only on one's browser). Still, it wasn't ideal and it could have posed a risk in the future, so it was addressed. `routeVars` was changed to use a `Map` instead of a plain object. `Map` isn't vulnerable.

Reply:
I think you can use `URLSearchParams` ([see my comment](https://github.com/writer/writer-framework/pull/554/files/57a2cabc643e7cfa92d952d2ecda31a65182db66#r1746547377)) which solve the issue